src/logger.ts

```diff
@@ -0,0 +1,66 @@
+import minimist from "minimist";
+import winston from "winston";
+
+export interface CustomLogger extends winston.Logger {
+    error: winston.LeveledLogMethod;
+    maker: winston.LeveledLogMethod;
+    taker: winston.LeveledLogMethod;
+    info: winston.LeveledLogMethod;
+    data: winston.LeveledLogMethod;
+    verbose: winston.LeveledLogMethod;
+}
+
+const colorizer = winston.format.colorize({
+    all: true,
+    colors: {
+        error: "red",
+        maker: "cyan",
+        taker: "yellow",
+        info: "purple",
+        data: "grey",
+        verbose: "green",
+    },
+});
+
+export function createLogger() {
+    const level = minimist(process.argv.slice(2), {
+        default: {
+            loglevel: "info",
+        },
+    }).loglevel;
+
+    if (
+        level !== "error" &&
```

Comment:
Are we actually using that many levels?

Reply:
Yes, but verbose can go away because it's only used once.

Reply:
Aiming for simple so I'd be keen to only have 3 levels at this stage:
- normal (info)
- With blockchain traces (trace or data)
- error (for errors :])

Reply:
Gonna remove the ability to set it to `verbose` _and_ `error`, because if you set it to `error` it's actually not even useful (you don't know when the swap has happened).